lib/ts/wrappers/express.ts

```diff
@@ -0,0 +1,268 @@
+/* Copyright (c) 2021, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Request, Response, NextFunction } from "express";
+import { HTTPMethod } from "../types";
+import { normaliseHttpMethod, sendNon200Response } from "../utils";
+import { BaseRequest } from "./request";
+import { BaseResponse } from "./response";
+import {
+    setCookieForServerResponse,
+    setHeaderForExpressLikeResponse,
+    getCookieValueFromIncomingMessage,
+    getHeaderValueFromIncomingMessage,
+    assertThatBodyParserHasBeenUsedForExpressLikeRequest,
+} from "./utils";
+import { Wrapper } from "./types";
+import NormalisedURLPath from "../normalisedURLPath";
+import RecipeModule from "../recipeModule";
+import { HEADER_RID } from "../constants";
+import STError from "../error";
+import SuperTokens from "../supertokens";
+import Session from "../recipe/session/recipe";
+import { SessionContainerInterface } from "../recipe/session/types";
+
+export class ExpressRequest extends BaseRequest {
+    private request: Request;
+    private parserChecked: boolean;
+
+    constructor(request: Request) {
+        super();
+        this.request = request;
+        this.parserChecked = false;
```

Comment:
Why not just call `assertThatBodyParserHasBeenUsedForExpressLikeRequest` in the constructor?

Reply:
Why not?

Reply:
let's say the API is related to file upload where a different parser is required. Even though the request reaches the middleware, having body parser check at the beginning of the middleware will be irrelevant (the request and response object are converted to BaseRequest and BaseResponse at the beginnging of middleware function)